subsystem/kubeapi_test.go

```diff
@@ -264,4 +362,31 @@ var _ = Describe("[kube-api]cluster installation", func() {
 			return conditionsv1.IsStatusConditionTrue(getAgentCRD(ctx, kubeClient, key).Status.Conditions, v1alpha1.AgentSyncedCondition)
 		}, "2m", "10s").Should(Equal(true))
 	})
+
+	It("WIP: deploy clusterDeployment and installEnv and verify updates", func() {
+		secretRef := deployLocalObjectSecretIfNeeded(ctx, kubeClient)
+		clusterDeploymentSpec := getDefaultClusterDeploymentSNOSpec(secretRef)
+		deployClusterDeploymentCRD(ctx, kubeClient, clusterDeploymentSpec)
+		clusterKubeName := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      clusterDeploymentSpec.ClusterName,
+		}
+		cluster := getClusterFromDB(ctx, kubeClient, db, clusterKubeName, waitForClusterReconcileTimeout)
+		waitForClusterDeploymentCRDState(ctx, kubeClient, clusterKubeName, models.ClusterStatusInsufficient, waitForClusterReconcileTimeout)
+		Expect(cluster.NoProxy).Should(Equal(""))
+		Expect(cluster.HTTPProxy).Should(Equal(""))
+		Expect(cluster.HTTPSProxy).Should(Equal(""))
+		Expect(cluster.AdditionalNtpSource).Should(Equal(""))
+
+		installEnvSpec := getDefaultInstallEnvSpec(secretRef, clusterDeploymentSpec)
+		deployInstallEnvCRD(ctx, kubeClient, "env", installEnvSpec)
+		waitForInstallEnvCRDState(ctx, kubeClient, clusterKubeName, models.ClusterStatusInsufficient, waitForClusterReconcileTimeout)
+		cluster = getClusterFromDB(ctx, kubeClient, db, clusterKubeName, waitForClusterReconcileTimeout)
+		waitForClusterDeploymentCRDState(ctx, kubeClient, clusterKubeName, models.ClusterStatusInsufficient, waitForClusterReconcileTimeout)
```

Comment:
if the state doesn't change then this wait will not really expect anything right? 
Maybe you can use `Eventually`?

Reply:
Done.